class/class-mainwp-custom-updater.php

```
@@ -0,0 +1,781 @@
+        if ( method_exists( $this, 'locate_installed_plugin_basename_recursive' ) ) {
+            $found = self::locate_installed_plugin_basename_recursive( $plugin, 600, 6 );
+        }
+        if ( ! $found && method_exists( $this, 'locate_installed_plugin_basename' ) && $slug ) {
+            $found = self::locate_installed_plugin_basename( $slug );
+        }
+
+        // final fallback: scan recently modified plugin files.
+        if ( ! $found ) {
+            $now         = time();
+            $all_plugins = get_plugins();
+            foreach ( $all_plugins as $basename => $data ) {
```

PHPMD (2.15.0)

Avoid unused local variables such as '$data'. (undefined)

(UnusedLocalVariable)